### ISSUE TITLE
Explicit runs when using NUnit-filters 'cat!=FOO'

### ISSUE THREAD

**Issue by OsirisTerje** [nunit/nunit3-vs-adapter#912]
I have a similar issue, but I use NUnit-filters 'cat!=FOO'

Using the enclosed test-project here, and adding some categorized filters, Explicit tests are executed
Running `dotnet test TestProject1.sln  -v:n --logger="Console;verbosity=detailed" -- NUnit.Where="cat!=FOO"`
products 

```
Running selected tests in C:\source\rune\NUnitTestProject1\TestProject1\bin\Debug\net5.0\TestProject1.dll
   NUnit3TestExecutor discovered 2 of 2 NUnit test cases using Current Discovery mode, Non-Explicit run
NUnit Adapter 4.1.0.0: Test execution complete
  Failed Test1 [30 ms]
  Stack Trace:
     at TestProject1.Tests.Test1() in C:\source\rune\NUnitTestProject1\TestProject1\UnitTest1.cs:line 12

  Passed TestBar [1 ms]
```
Adding 'cat!=Explicit' does not help

(btw, using legacy discovery does help, and we will use that for now)

All tests:
```
[Test]
[Explicit]
public void Test1()
{
	Assert.Fail();
}

[Test]
[Category("FOO")]
public void TestFoo()
{
    Assert.Pass();
}

[Test]
[Category("BAR")]
public void TestBar()
{
    Assert.Pass();
}
```

_Originally posted by @runehalfdan in https://github.com/nunit/nunit3-vs-adapter/issues/874#issuecomment-964263589_

**Comment by OsirisTerje** [comment]
@runehalfdan  Version 4.2.0-beta.1 released now on nuget.org  https://www.nuget.org/packages/NUnit3TestAdapter/4.2.0-beta.1 .  It also contains an engine upgrade, which is the reason for the beta.  Would appreciate confirmation that it still works for you.

**Comment by runehalfdan** [comment]
@OsirisTerje The fix works in this test-project

As I understand, some test-discovery issues has also been fixed in the 4.2.0. beta too, so will take some time to verify the beta in our codebase.
